torch/csrc/jit/codegen/cuda/kernel.cpp

```diff
@@ -71,13 +71,19 @@ class KernelIrScanner : private kir::IrVisitor {
     // Do we have any reductions?
     summary_.has_block_reductions =
         summary_.has_block_reductions || domain->hasBlockReduction();
-    summary_.has_grid_reductions =
-        summary_.has_grid_reductions || domain->hasGridReduction();
 
     // Do we have block broadcasts?
     summary_.has_block_broadcasts =
         summary_.has_block_broadcasts || domain->hasBlockBroadcast();
 
+    if (domain->hasGridReduction()) {
+      // tensor_index may be for initialization of a reduction
+      // buffer. Avoid counting twice.
+      if (tensor_index->definition()->isA<kir::ReductionOp>()) {
```

Comment:
did you mean `if (`tensor_index->definition()->isA<kir::ReductionOp>())` (note the logical not)